server/route/MessagingController.js

```diff
@@ -0,0 +1,201 @@
+					connectedUsers[data.username] = ws; // Store the WebSocket connection
+					console.log(`${data.username} joined the room`);
+				} else if (data.event === "sendMessage") {
+					const { clientUsername, stylistUsername, sender, content } =
+						data;
+
+					// Validate data
+					if (
+						!clientUsername ||
+						!stylistUsername ||
+						!sender ||
+						!content
+					) {
+						console.error("Invalid message data");
+						ws.send(
+							JSON.stringify({
+								event: "error",
+								message: "Invalid message data",
+							})
+						);
+						return;
+					}
+
+					// Save the message to MongoDB
+					const newMessage = new Message({
+						clientUsername,
+						stylistUsername,
+						sender,
+						content,
+					});
+					await newMessage.save();
+
+					// Find the recipient WebSocket connection
+					const recipientUsername =
+						sender === clientUsername
+							? stylistUsername
+							: clientUsername;
+					const recipientWs = connectedUsers[recipientUsername];
+
+					if (
+						recipientWs &&
+						recipientWs.readyState === WebSocketServer.OPEN
+					) {
+						recipientWs.send(
+							JSON.stringify({
+								event: "receiveMessage",
+								clientUsername,
+								stylistUsername,
+								sender,
+								content,
+								createdAt: newMessage.createdAt,
+							})
+						);
+					}
+
+					console.log(
+						`Message from ${sender} sent to ${recipientUsername}`
+					);
+				}
+			} catch (error) {
+				console.error("Error handling message:", error);
+			}
+		});
+
+		ws.on("close", () => {
+			console.log("WebSocket connection closed");
+			if (ws.username) {
+				delete connectedUsers[ws.username]; // Remove the disconnected user
+			}
+		});
+	});
+};
+
+// Endpoint to fetch chat history between a client and a stylist
+router.get(
+	"/history/:clientUsername/:stylistUsername",
+	asyncHandler(async (req, res) => {
+		const { clientUsername, stylistUsername } = req.params;
+
+		// Fetch chat history
+		const messages = await Message.find({
+			clientUsername,
+			stylistUsername,
+		}).sort({ createdAt: 1 });
+
+		// Format the response for the chat page
+		const formattedMessages = messages.map((message) => ({
+			content: message.content,
+			sender: message.sender,
+			createdAt: message.createdAt,
+		}));
+
+		// Fetch the profile picture and name for the stylist
+		const stylist = await Stylist.findOne({ username: stylistUsername });
+		const client = await Client.findOne({ username: clientUsername });
+
+		res.json({
+			stylist: {
+				name: stylist?.info?.name,
+				profilePicture: stylist?.profilePhoto?.photoData,
+			},
+			client: {
+				name: client?.info?.name,
+				profilePicture: client?.profilePhoto?.photoData,
+			},
+			messages: formattedMessages,
+		});
+	})
+);
+
+// Endpoint to fetch recent messages for the Messages Page
+router.get(
+	"/recent/:username",
+	asyncHandler(async (req, res) => {
+		const { username } = req.params;
```

Comment:
Similar points as before:
1. Might be cleaner/more RESTful to use query params here
2. Validate that the account with the `username` exists in accounts